front_end/Profile/UserProfile.js

```diff
@@ -0,0 +1,178 @@
+                    <Text>Cancel</Text>
+                </TouchableOpacity>
+            ),
+            headerRight: () => (
+                <TouchableOpacity onPress={this.onPressSave}>
+                    <Text>Save</Text>
+                </TouchableOpacity>
+            )
+        });
+    }
+  
+    toggleInterest() {
+        let temp = this.state.editState;
+        temp.interestEnabled = !temp.interestEnabled;
+        this.setState({ editState: temp });
+    }
+  
+    toggleActivity() {
+        let temp = this.state.editState;
+        temp.activityEnabled = !temp.activityEnabled;
+        this.setState({ editState: temp });
+    }
+  
+    handleChange(value, key) {
+        let temp = this.state.editState;
+        temp[key] = key != "age" ? value : value.replace(/[^0-9]/g, '');
+        this.setState({ editState: temp });
+    }
+
+    render() {
+        return (
+            <SafeAreaView style={pro_styles.background}>
+                <ScrollView>
+                    <View style={{ alignItems: 'center' }}>
+                        <Icon name='profile' size={160} />
+                        {this.state.isOwner && !this.state.isEditing &&
+                            <TouchableOpacity style={pro_styles.edit_button} onPress={this.onPressEdit}>
+                                <Text style={styles.location}>Edit Profile</Text>
+                            </TouchableOpacity>
+                        }
+                        {this.state.isEditing ?
+                            <Headline 
+                                isEdit={true}
+                                name={this.state.editState.username}
+                                location={this.state.editState.location}
+                                occupation={this.state.editState.occupation}
+                                age={this.state.editState.age}
+                                handleChange={this.handleChange}
+                            /> :
+                            <Headline 
+                                isEdit={false}
+                                name={this.state.username}
+                                location={this.state.location}
+                                occupation={this.state.occupation}
+                                age={this.state.age}
+                                handleChange={this.handleChange}
+                            />
+                        }
+                        {this.state.isEditing ?
+                            <Bio
+                                isEditing={true}
+                                about={this.state.editState.about}
+                                verse={this.state.verse}
+                                handleChange={this.handleChange}
+                                interestEnabled={this.state.editState.interestEnabled}
+                                activityEnabled={this.state.editState.activityEnabled}
+                                interests={this.state.editState.interests}
```

Comment:
I didn't mention this in earlier PRs but with a more complicated class like this, I think it is better to extract the variables first. 

```
{editState, verse} = this.state;
{interestEnabled, activityEnabled, interests} = editState;
return ({interestEnabled} {activityEnabled} whateverelse);
```